modules/trainer/GenericTrainer.py

```diff
@@ -141,6 +143,17 @@ def start(self):
 
         self.parameters = list(self.model_setup.create_parameters(self.model, self.config))
 
+    @functools.cached_property
+    def __adaptive_optimizer(self):
```

Comment:
Please move this function into the Optimizer enum. Similar to all the functions in the ModelType enum

Reply:
Done.